root/static/scripts/relationship-editor/components/DialogLinkType.js

```diff
@@ -278,6 +281,12 @@ const DialogLinkType = (React.memo<PropsT>(({
     });
   }, [dispatch, source]);
 
+  const entityType0 = autocomplete.items[0].entity?.type0 ||
+    autocomplete.items[1].entity?.type0;
+  const entityType1 = autocomplete.items[0].entity?.type1 ||
+    autocomplete.items[1].entity?.type1;
+  const linkType = autocomplete.selectedItem?.entity;
```

Comment:
This should be a more reliable way of getting type0/type1:
```suggestion
  const linkType = autocomplete.selectedItem?.entity;
  const entityType0 = linkType?.type0;
  const entityType1 = linkType?.type1;
```

Reply:
Except that `entityType0` and `entityType1` are for use when `linkType` is not defined as no relationship type is selected, see the second commit’s message and bonus in pull request’s description.

Reply:
In that case I think I'd pass `targetType` to `DialogLinkType` since we already have the source type too.

Reply:
Is there another place where the two types are alphabetically ordered already? It seems to be in `LinkTypeAutocomplete`’s `id` (according to React Developer Tools) but I failed to trace it back in the code.

Reply:
I think the link type ID comes from `getRelationshipKey`.  In a few other places I see we just sort the source/target types, like in `getDialogLinkTypeOptions`, `hasDialogLinkTypeOptions`, and `editorMayEditTypes` in exportTypeInfo.js

Reply:
Thank you it is much cleaner now; See https://github.com/metabrainz/musicbrainz-server/pull/2866/commits/5cc69d12fc807389451d69d98bed8ae5979a440b?diff=split&w=1